tests/component/component.cpp

```diff
@@ -45,16 +45,193 @@ TEST(Component, unset_name) {
+    const std::string e2 = "<component/><component name=\"child2\"/><component name=\"child1\"/><encapsulation><component_ref><component_ref component=\"child2\"/><component_ref component=\"child1\"/></component_ref></encapsulation>";
+    libcellml::Component c;
+    libcellml::Component_Ptr c1 = std::make_shared<libcellml::Component>();
+    libcellml::Component_Ptr c2 = std::make_shared<libcellml::Component>();
+    c1->setName("child1");
+    c2->setName("child2");
+    c.addComponent(c1);
+    c.addComponent(c2);
+
+    c.removeComponent(0);
+    EXPECT_EQ(1, c.componentCount());
+    std::string a = c.serialise(libcellml::CELLML_FORMAT_XML);
+    EXPECT_EQ(e1, a);
+    EXPECT_THROW(c.removeComponent(1), std::out_of_range);
+
+    c.addComponent(c1);
+    c.addComponent(c1);
+    // Remove the first occurence of "child1".
+    c.removeComponent("child1");
+    EXPECT_EQ(2, c.componentCount());
+    a = c.serialise(libcellml::CELLML_FORMAT_XML);
+    EXPECT_EQ(e2, a);
+    EXPECT_THROW(c.removeComponent("child3"), std::out_of_range);
+}
+
+TEST(Component, getcomponent) {
+    const std::string e1 = "<component/><component name=\"childA\"/><encapsulation><component_ref><component_ref component=\"childA\"/></component_ref></encapsulation>";
+    const std::string e2 = "<component name=\"parent\"/><component name=\"gus\"/><component name=\"childB\"/><component name=\"child3\"/><component name=\"gus\"/><component name=\"childB\"/><component name=\"child3\"/><encapsulation><component_ref component=\"parent\"><component_ref component=\"gus\"><component_ref component=\"childB\"/><component_ref component=\"child3\"/></component_ref><component_ref component=\"gus\"><component_ref component=\"childB\"/><component_ref component=\"child3\"/></component_ref></component_ref></encapsulation>";
+    libcellml::Component c;
+    libcellml::Component_Ptr c1 = std::make_shared<libcellml::Component>();
+    libcellml::Component_Ptr c2 = std::make_shared<libcellml::Component>();
+    libcellml::Component_Ptr c3 = std::make_shared<libcellml::Component>();
+    c1->setName("child1");
+    c2->setName("child2");
+    c3->setName("child3");
+    c.addComponent(c1);
+
+    libcellml::Component_Ptr cA = c.getComponent(0);
+    cA->setName("childA");
+
+    std::string a = c.serialise(libcellml::CELLML_FORMAT_XML);
+    EXPECT_EQ(e1, a);
+
+    // Using const version of overloaded method
+    const libcellml::Component_Ptr cS = static_cast<const libcellml::Component>(c).getComponent(0);
+    // Can do this as we just have a const pointer
+    cS->setName("gus");
+    EXPECT_EQ("gus", cS->getName());
+    EXPECT_THROW(c.getComponent(4), std::out_of_range);
+
+    libcellml::Component_Ptr cAr = c.getComponent("gus");
+    EXPECT_EQ("gus", cAr->getName());
+
+    // Modify a deeper Component
+    c.setName("parent");
+    c.addComponent(c1);
+    c1->addComponent(c2);
+    c1->addComponent(c3);
```

Comment:
Same here, not sure why ```c1``` is now added to ```c``` before ```c2``` and ```c3``` are added to ```c1```. Just 'cleaning up' the code?